worker/src/RTC/RtpPacket.cpp

```diff
@@ -650,20 +650,23 @@ namespace RTC
 		}
 	}
 
+	/**
+	 * NOTE: This method automatically adjusts padding for padding+payload to be
+	 * padded to 4 bytes.
+	 */
 	void RtpPacket::SetPayloadLength(size_t length)
 	{
 		MS_TRACE();
 
-		// Pad desired length to 4 bytes.
-		length = static_cast<size_t>(Utils::Byte::PadTo4Bytes(static_cast<uint16_t>(length)));
+		auto payloadPadding = Utils::Byte::PadTo4Bytes(static_cast<uint16_t>(length)) - length;
 
 		this->size -= this->payloadLength;
 		this->size -= size_t{ this->payloadPadding };
 		this->payloadLength  = length;
-		this->payloadPadding = 0u;
-		this->size += length;
+		this->payloadPadding = payloadPadding;
+		this->size += this->payloadLength + this->payloadPadding;
 
-		SetPayloadPaddingFlag(false);
+		SetPayloadPaddingFlag(this->payloadPadding > 0);
```

Comment:
Oh this is not correct because we have to write number of padding bytes in the last byte of the padding right? I mean, SerPayloadPaddingFlag() doesn't do it AFAIR (I am out now).